### ISSUE TITLE
nrpe_user directive doesn't work as expected?

### ISSUE THREAD

**Issue by david-cimburek** [NagiosEnterprises/nrpe#40]
It seems to me that **nrpe_user** directive in **nrpe.cfg** doesn't work as supposed for nrpe-3.0-beta1. I have a 64bit Gentoo Linux distribution. When I build nrpe from latest sources and configure

```
nrpe_user=nagios
command[check_id]=/usr/bin/id
```

and start nrpe via default init script it seems that the nrpe daemon is running with nagios user privileges - at least according to list of system processes:

```
$ ps -ef | grep nrpe
nagios   14141     1  0 06:44 ?        00:00:00 /usr/bin/nrpe -c /etc/nrpe.cfg -d

```
But when I call the prepared **check_id** command (`check_nrpe -H localhost -c check_id`) I get **root** UID and nagios GID:

`uid=0(root) gid=1002(nagios) groups=1002(nagios)`

I would expect that I will get **nagios** UID. Am I doing something wrong or is there really some bug in nrpe?

**Comment by jfrickson** [comment]
I created a `check_id` script like this:
    #!/bin/sh
    id
Added `command[check_id]=/usr/local/nagios/lib/check_id` to `nrpe.cfg` then ran:
    john ~ $ /usr/local/nagios/bin/check_nrpe -H 127.0.0.1 -c check_id
    uid=500(nagios) gid=101(nagios) groups=101(nagios),8(www)
It looks fine here. What does your `check_id` look like?

**Comment by david-cimburek** [comment]
I did run the test again with the latest GIT version and the result is still the same.

If I start the nrpe daemon as a nagios user than the UID returned by check_id command is correct. But if nrpe is started by system init script (=by root user) then check_id still keeps the root's UID 0.

Did you try the scenario with starting the nrpe daemon as a root user?

**Comment by jfrickson** [comment]
Ok, I found the problem. Fixed now in the nrpe-3.0-RC2 release (https://github.com/NagiosEnterprises/nrpe/tree/nrpe-3.0-RC2).

See commits:
https://github.com/NagiosEnterprises/nrpe/commit/a3dd5d92178c8d14cbab4c7cf5fa3adcb5f2e645
https://github.com/NagiosEnterprises/nrpe/commit/6e6688aac36447244c23b3c4b638bc7e4669bed3

**Comment by david-cimburek** [comment]
Great, I confirm that the problem is fixed. Thank you!
